### ISSUE TITLE
RS^3Mamba: test question

### ISSUE THREAD

**Issue by derow0208** [sstary/SSRS#58]
in `RS3Mamba/train_Mamba.py`, 
```
                # Fill in the results array
                for out, (x, y, w, h) in zip(outs, coords):
                    out = out.transpose((1, 2, 0))
                    pred[x:x + w, y:y + h] += out
```
however, the `window_size=(256,256)` and `strides=128` in the train function, which is `MIoU = test(net, test_ids, all=False, stride=Stride_Size)`. Will the `pred[x:x + w, y:y + h] += out` cause overlapping problems? should i change the strides to be equal to window_size[0]?

**Comment by derow0208** [comment]
to be specific, in `sliding_window(img, step=stride, window_size=window_size)`, it will cause overlaping problems.

**Comment by sstary** [comment]
The sliding window size is set to 256 × 256 with a big stride in the training stage and a small stride in the test stage. A smaller stride in the test stage can reduce border effects by averaging the prediction results on the overlapping regions.

**Comment by derow0208** [comment]
At training stage, the loss was calculated without considering the border effects, `loss = loss_calc(output, target, weights)`. u just considered the overlaping while calculating the mIoU. will the difference between loss calculating and mIoU calculating affect the ckpt saving?

**Comment by sstary** [comment]
The loss is calculated regardless of the stride size. In other words, none of the testing phase has any impact on the training phase. The test during training is only to record all the models and select the best one. In the test phase, smaller steps are just for better MIoU and visualization.
